core/vm/operations_verkle.go

```diff
@@ -97,17 +97,20 @@ var (
 
 func gasSelfdestructEIP4762(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
 	beneficiaryAddr := common.Address(stack.peek().Bytes20())
-	if _, isPrecompile := evm.precompile(beneficiaryAddr); isPrecompile {
-		return 0, nil
-	}
-
 	contractAddr := contract.Address()
+
 	statelessGas := evm.Accesses.TouchBasicData(contractAddr[:], false)
```

Comment:
While it's true that this touch should have not practical effect, I'd still keep it after the precompile if.